scipy/spatial/transform/_rotation.pyx

```diff
@@ -223,6 +230,131 @@ cdef double[:, :] _compute_euler_from_matrix(
 
     return angles
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef double[:, :] _compute_euler_from_quat(
+    np.ndarray[double, ndim=2] quat, const uchar[:] seq, bint extrinsic=False
+):
+    # The algorithm assumes intrinsic frame transformations. The algorithm
+    # in the paper is formulated for rotation matrices which are transposition
+    # rotation matrices used within Rotation.
+    # Adapt the algorithm for our case by
+    # 1. Instead of transposing our representation, use the transpose of the
+    #    O matrix as defined in the paper, and be careful to swap indices
+    # 2. Reversing both axis sequence and angles for extrinsic rotations
+    
+    if not extrinsic:
+        seq = seq[::-1]
+
+    cdef int i = _elementary_basis_index(seq[0])
+    cdef int j = _elementary_basis_index(seq[1])
+    cdef int k = _elementary_basis_index(seq[2])
+
+    is_proper = i == k
```

Comment:
I believe type `cdef bint` might be used here for consistency with other code.

Reply:
Alright! So I should change it to
```
cdef bint i = _elementary_basis_index(seq[0])
cdef bint j = _elementary_basis_index(seq[1])
cdef bint k = _elementary_basis_index(seq[2])
```

Reply:
`cdef bint` is for boolean values though, isn't it? In this case it should be `int`, since `i`, `j` and `k` can be either 0, 1 or 2.

Reply:
I meant for `is_proper`.

Reply:
Ah of course! Sorry about the confusion!